# HG changeset patch
# User Rik <rik@octave.org>
# Date 1354556338 28800
#      Mon Dec 03 09:38:58 2012 -0800
# Node ID 1a800034d4435df0f0e05ac326785cf94d6a8573
# Parent  20e9b56bbf2f627eb0f58d0e2fea7238a1c1c83b
Colormap error() calls now report the input variable which is a problem.

* autumn.m, bone.m, colorcube.m, cool.m, copper.m, flag.m, gmap40.m, gray.m,
hot.m, hsv.m, jet.m, lines.m, ocean.m, pink.m, prism.m, rainbow.m, spring.m,
summer.m, white.m, winter.m: Colormap error() calls now report the input
variable which is a problem.

diff --git a/scripts/image/autumn.m b/scripts/image/autumn.m
--- a/scripts/image/autumn.m
+++ b/scripts/image/autumn.m
@@ -32,17 +32,17 @@
 ## PKG_DEL: colormap ("unregister", "autumn");
 
 function map = autumn (n)
 
   if (nargin == 0)
     n = rows (colormap);
   elseif (nargin == 1)
     if (! isscalar (n))
-      error ("autumn: argument must be a scalar");
+      error ("autumn: N must be a scalar");
     endif
   else
     print_usage ();
   endif
 
   if (n == 1)
     map = [1, 0, 0];
   elseif (n > 1)
diff --git a/scripts/image/bone.m b/scripts/image/bone.m
--- a/scripts/image/bone.m
+++ b/scripts/image/bone.m
@@ -32,17 +32,17 @@
 ## PKG_DEL: colormap ("unregister", "bone");
 
 function map = bone (n)
 
   if (nargin == 0)
     n = rows (colormap);
   elseif (nargin == 1)
     if (! isscalar (n))
-      error ("bone: argument must be a scalar");
+      error ("bone: N must be a scalar");
     endif
   else
     print_usage ();
   endif
 
   if (n == 1)
     map = [0.125, 0.125, 0.125];
   elseif (n > 1)
diff --git a/scripts/image/colorcube.m b/scripts/image/colorcube.m
--- a/scripts/image/colorcube.m
+++ b/scripts/image/colorcube.m
@@ -30,17 +30,17 @@
 ## @end deftypefn
 
 function map = colorcube (n)
 
   if (nargin == 0)
     n = rows (colormap);
   elseif (nargin == 1)
     if (! isscalar (n))
-      error ("colorcube: argument must be a scalar");
+      error ("colorcube: N must be a scalar");
     endif
   else
     print_usage ();
   endif
 
   if (n < 9)
     map = gray (n);
     return;
diff --git a/scripts/image/cool.m b/scripts/image/cool.m
--- a/scripts/image/cool.m
+++ b/scripts/image/cool.m
@@ -31,17 +31,17 @@
 ## PKG_DEL: colormap ("unregister", "cool");
 
 function map = cool (n)
 
   if (nargin == 0)
     n = rows (colormap);
   elseif (nargin == 1)
     if (! isscalar (n))
-      error ("cool: argument must be a scalar");
+      error ("cool: N must be a scalar");
     endif
   else
     print_usage ();
   endif
 
   if (n == 1)
     map = [0, 1, 1];
   elseif (n > 1)
diff --git a/scripts/image/copper.m b/scripts/image/copper.m
--- a/scripts/image/copper.m
+++ b/scripts/image/copper.m
@@ -32,17 +32,17 @@
 ## PKG_DEL: colormap ("unregister", "copper");
 
 function map = copper (n)
 
   if (nargin == 0)
     n = rows (colormap);
   elseif (nargin == 1)
     if (! isscalar (n))
-      error ("copper: argument must be a scalar");
+      error ("copper: N must be a scalar");
     endif
   else
     print_usage ();
   endif
 
   if (n == 1)
     map = [0, 0, 0];
   elseif (n > 1)
diff --git a/scripts/image/flag.m b/scripts/image/flag.m
--- a/scripts/image/flag.m
+++ b/scripts/image/flag.m
@@ -32,17 +32,17 @@
 ## PKG_DEL: colormap ("unregister", "flag");
 
 function map = flag (n)
 
   if (nargin == 0)
     n = rows (colormap);
   elseif (nargin == 1)
     if (! isscalar (n))
-      error ("flag: argument must be a scalar");
+      error ("flag: N must be a scalar");
     endif
   else
     print_usage ();
   endif
 
   if (n == 1)
     map = [1, 0, 0];
   elseif (n > 1)
diff --git a/scripts/image/gmap40.m b/scripts/image/gmap40.m
--- a/scripts/image/gmap40.m
+++ b/scripts/image/gmap40.m
@@ -32,17 +32,17 @@
 ## PKG_DEL: colormap ("unregister", "gmap40");
 
 function map = gmap40 (n)
 
   if (nargin == 0)
     n = 6;
   elseif (nargin == 1)
     if (! isscalar (n))
-      error ("gmap40: argument must be a scalar");
+      error ("gmap40: N must be a scalar");
     endif
   else
     print_usage ();
   endif
 
   if (n > 1)
     C = [1, 0, 0; 0, 1, 0; 0, 0, 1; 1, 1, 0; 1, 0, 1; 0, 1, 1];
     map = C(rem (0:(n-1), 6) + 1, :);
diff --git a/scripts/image/gray.m b/scripts/image/gray.m
--- a/scripts/image/gray.m
+++ b/scripts/image/gray.m
@@ -34,17 +34,17 @@
 ## PKG_DEL: colormap ("unregister", "gray");
 
 function map = gray (n)
 
   if (nargin == 0)
     n = rows (colormap);
   elseif (nargin == 1)
     if (! isscalar (n))
-      error ("gray: argument must be a scalar");
+      error ("gray: N must be a scalar");
     endif
   else
     print_usage ();
   endif
 
   if (n == 1)
     map = [0, 0, 0];
   elseif (n > 1)
diff --git a/scripts/image/hot.m b/scripts/image/hot.m
--- a/scripts/image/hot.m
+++ b/scripts/image/hot.m
@@ -32,17 +32,17 @@
 ## PKG_DEL: colormap ("unregister", "hot");
 
 function map = hot (n)
 
   if (nargin == 0)
     n = rows (colormap);
   elseif (nargin == 1)
     if (! isscalar (n))
-      error ("hot: argument must be a scalar");
+      error ("hot: N must be a scalar");
     endif
   else
     print_usage ();
   endif
 
   if (n == 1)
     map = [1, 1, 1];
   elseif (n > 1)
diff --git a/scripts/image/hsv.m b/scripts/image/hsv.m
--- a/scripts/image/hsv.m
+++ b/scripts/image/hsv.m
@@ -36,17 +36,17 @@
 ## PKG_DEL: colormap ("unregister", "hsv");
 
 function map = hsv (n)
 
   if (nargin == 0)
     n = rows (colormap);
   elseif (nargin == 1)
     if (! isscalar (n))
-      error ("hsv: argument must be a scalar");
+      error ("hsv: N must be a scalar");
     endif
   else
     print_usage ();
   endif
 
   if (n == 1)
     map = [1, 0, 0];
   elseif (n > 1)
diff --git a/scripts/image/jet.m b/scripts/image/jet.m
--- a/scripts/image/jet.m
+++ b/scripts/image/jet.m
@@ -32,17 +32,17 @@
 ## PKG_DEL: colormap ("unregister", "jet");
 
 function map = jet (n)
 
   if (nargin == 0)
     n = rows (colormap);
   elseif (nargin == 1)
     if (! isscalar (n))
-      error ("jet: argument must be a scalar");
+      error ("jet: N must be a scalar");
     endif
   else
     print_usage ();
   endif
 
   if (n == 1)
     map = [0, 1, 1];
   elseif (n > 1)
diff --git a/scripts/image/lines.m b/scripts/image/lines.m
--- a/scripts/image/lines.m
+++ b/scripts/image/lines.m
@@ -31,17 +31,17 @@
 ## PKG_DEL: colormap ("unregister", "lines");
 
 function map = lines (n)
 
   if (nargin == 0)
     n = rows (colormap);
   elseif (nargin == 1)
     if (! isscalar (n))
-      error ("lines: argument must be a scalar");
+      error ("lines: N must be a scalar");
     endif
   else
     print_usage ();
   endif
 
   if (n == 1)
     map = [0, 0, 1];
   elseif (n > 1)
diff --git a/scripts/image/ocean.m b/scripts/image/ocean.m
--- a/scripts/image/ocean.m
+++ b/scripts/image/ocean.m
@@ -34,17 +34,17 @@
 ## PKG_DEL: colormap ("unregister", "ocean");
 
 function map = ocean (n)
 
   if (nargin == 0)
     n = rows (colormap);
   elseif (nargin == 1)
     if (! isscalar (n))
-      error ("ocean: argument must be a scalar");
+      error ("ocean: N must be a scalar");
     endif
   else
     print_usage ();
   endif
 
   if (n == 1)
     map = [0, 0, 0];
   elseif (n > 1)
diff --git a/scripts/image/pink.m b/scripts/image/pink.m
--- a/scripts/image/pink.m
+++ b/scripts/image/pink.m
@@ -32,17 +32,17 @@
 ## PKG_DEL: colormap ("unregister", "pink");
 
 function map = pink (n)
 
   if (nargin == 0)
     n = rows (colormap);
   elseif (nargin == 1)
     if (! isscalar (n))
-      error ("pink: argument must be a scalar");
+      error ("pink: N must be a scalar");
     endif
   else
     print_usage ();
   endif
 
   if (n == 1)
     map = sqrt ([1/3, 1/3, 1/3]);
   elseif (n > 1)
diff --git a/scripts/image/prism.m b/scripts/image/prism.m
--- a/scripts/image/prism.m
+++ b/scripts/image/prism.m
@@ -32,17 +32,17 @@
 ## PKG_DEL: colormap ("unregister", "prism");
 
 function map = prism (n)
 
   if (nargin == 0)
     n = rows (colormap);
   elseif (nargin == 1)
     if (! isscalar (n))
-      error ("prism: argument must be a scalar");
+      error ("prism: N must be a scalar");
     endif
   else
     print_usage ();
   endif
 
   if (n == 1)
     map = [1 0 0];
   elseif (n > 1)
diff --git a/scripts/image/rainbow.m b/scripts/image/rainbow.m
--- a/scripts/image/rainbow.m
+++ b/scripts/image/rainbow.m
@@ -35,17 +35,17 @@
 ## PKG_DEL: colormap ("unregister", "rainbow");
 
 function map = rainbow (n)
 
   if (nargin == 0)
     n = rows (colormap);
   elseif (nargin == 1)
     if (! isscalar (n))
-      error ("rainbow: argument must be a scalar");
+      error ("rainbow: N must be a scalar");
     endif
   else
     print_usage ();
   endif
 
   if (n == 1)
     map = [1, 0, 0];
   elseif (n > 1)
diff --git a/scripts/image/spring.m b/scripts/image/spring.m
--- a/scripts/image/spring.m
+++ b/scripts/image/spring.m
@@ -31,17 +31,17 @@
 ## PKG_DEL: colormap ("unregister", "spring");
 
 function map = spring (n)
 
   if (nargin == 0)
     n = rows (colormap);
   elseif (nargin == 1)
     if (! isscalar (n))
-      error ("spring: argument must be a scalar");
+      error ("spring: N must be a scalar");
     endif
   else
     print_usage ();
   endif
 
   if (n == 1)
     map = [1, 0, 1];
   elseif (n > 1)
diff --git a/scripts/image/summer.m b/scripts/image/summer.m
--- a/scripts/image/summer.m
+++ b/scripts/image/summer.m
@@ -32,17 +32,17 @@
 ## PKG_DEL: colormap ("unregister", "summer");
 
 function map = summer (n)
 
   if (nargin == 0)
     n = rows (colormap);
   elseif (nargin == 1)
     if (! isscalar (n))
-      error ("summer: argument must be a scalar");
+      error ("summer: N must be a scalar");
     endif
   else
     print_usage ();
   endif
 
   if (n == 1)
     map = [0, 0.5, 0.4];
   elseif (n > 1)
diff --git a/scripts/image/white.m b/scripts/image/white.m
--- a/scripts/image/white.m
+++ b/scripts/image/white.m
@@ -31,17 +31,17 @@
 ## PKG_DEL: colormap ("unregister", "white");
 
 function map = white (n)
 
   if (nargin == 0)
     n = rows (colormap);
   elseif (nargin == 1)
     if (! isscalar (n))
-      error ("white: argument must be a scalar");
+      error ("white: N must be a scalar");
     endif
   else
     print_usage ();
   endif
 
   map = ones (n, 3);
 
 endfunction
diff --git a/scripts/image/winter.m b/scripts/image/winter.m
--- a/scripts/image/winter.m
+++ b/scripts/image/winter.m
@@ -31,17 +31,17 @@
 ## PKG_DEL: colormap ("unregister", "winter");
 
 function map = winter (n)
 
   if (nargin == 0)
     n = rows (colormap);
   elseif (nargin == 1)
     if (! isscalar (n))
-      error ("winter: argument must be a scalar");
+      error ("winter: N must be a scalar");
     endif
   else
     print_usage ();
   endif
 
   if (n == 1)
     map = [0, 0, 1];
   elseif (n > 1)
